dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -25,52 +25,88 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 import org.eclipse.aether.artifact.Artifact;
 import org.eclipse.aether.graph.Dependency;
 
 /**
- * Path from the root to a node in a dependency tree, where a node is a dependency.
+ * Path from the root to a node in a dependency tree, where the root node is a Maven artifact and
```

Comment:
It occurs to me here that paths don't have roots. Can we rewrite this as simply a path without reference to dependency trees? We can still use it in the dependency tree, but this class doesn't need to know that.

Reply:
Updated as `A sequence of Maven artifacts and dependency (scope and optional flag) in between.`